keps/sig-cli/4764-kubectl-watchlist/README.md

```diff
@@ -0,0 +1,757 @@
+  - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Kubectl supports WatchList to list resources. This will allow requests from kubectl to hit the WatchCache,
+giving a performance boost and reducing impact on the kube-apiserver.
+
+## Motivation
+
+Currently `client-go` could enable `WatchList` by env and it works great, but kubectl is still a pain. 
+`kubectl` doest not have a cache mechanism like `client-go`, so it will do `list`  whenever it is used. Furthermore,
+request with pager params can not hit Cache of `kube-apiserver`, every request will go through into `etcd`,
+which could push up the load of `kube-apiserver` and make system unstable.
+
+In practice, our users have been writing script to unrestrainedly call `kubectl`, which could lead serious performance
+issue. As cluster admininstor, we have few method to eliminate it. APF maybe is an option, but it's hard to configure
+and don't solve the problem from the root. In the other side, if we are standing in the shoes of user, most of those 
+scripts are simple Bash or Python which just simply glue a few `kubectl` commands. It is too heavy and "not worth" 
+for user to refactor just "a few commands" into "a batch of complex code". So we are seeking an optimization for `kubectl`. With this optimization, all users of kubectl can get free performance improvement.
+
+
+
+### Goals
+
+* Introduce WatchList to `kubectl` without breaking any user interface.
+* Make `kubectl` hit the cache, thus reducing impact on `kube-apiserver`.
+
+### Non-Goals
+
+* Change kubectl functionality.
+
+## Proposal
+
+`kubectl` commands like `get`, `describe` now should be able to use `WatchList` to list resources, rather than
```

Comment:
```suggestion
Existing `kubectl` subcommands like `get`, `describe` should be able to use a _watch list_ to fetch collections (list resources), rather than
```